scripts/analytics/node_count.py

```diff
@@ -0,0 +1,113 @@
+from keen.client import KeenClient
+
+logger = logging.getLogger(__name__)
+logging.basicConfig(level=logging.INFO)
+
+
+def count(end_date=None):
+    node_query = (
+        Q('is_deleted', 'ne', True) &
+        Q('is_folder', 'ne', True)
+    )
+
+    if end_date:
+        node_query = node_query & Q('date_created', 'lt', end_date)
+
+    registration_query = node_query & Q('is_registration', 'eq', True)
+    non_registration_query = node_query & Q('is_registration', 'eq', False)
+    project_query = non_registration_query & Q('parent_node', 'eq', None)
+    registered_project_query = registration_query & Q('parent_node', 'eq', None)
+    public_query = Q('is_public', 'eq', True)
+    private_query = Q('is_public', 'eq', False)
+    retracted_query = Q('retraction', 'ne', None)
+    node_public_query = non_registration_query & public_query
+    node_private_query = non_registration_query & private_query
+    project_public_query = project_query & public_query
+    project_private_query = project_query & private_query
+    registered_node_public_query = registration_query & public_query
+    registered_node_private_query = registration_query & private_query
+    registered_node_retracted_query = registration_query & retracted_query
+    registered_project_public_query = registered_project_query & public_query
+    registered_project_private_query = registered_project_query & private_query
+    registered_project_retracted_query = registered_project_query & retracted_query
+
+    totals = {
+        'nodes': {
+            'total': Node.find(node_query).count(),
+        },
+        'projects': {
+            'total': Node.find(project_query).count(),
+        },
+        'registered_nodes': {
+            'total': Node.find(registration_query).count(),
+        },
+        'registered_projects': {
+            'total': Node.find(registered_project_query).count(),
+        }
+    }
+
+    if not end_date:
+        totals['nodes'].update({
+            'public': Node.find(node_public_query).count(),
+            'private': Node.find(node_private_query).count()
+        })
+
+        totals['projects'].update({
+            'public': Node.find(project_public_query).count(),
+            'private': Node.find(project_private_query).count(),
+        })
+
+        totals['registered_nodes'].update({
+            'public': Node.find(registered_node_public_query).count(),
+            'embargoed': Node.find(registered_node_private_query).count(),
+            'withdrawn': Node.find(registered_node_retracted_query).count(),
+        })
+
+        totals['registered_projects'].update({
+            'public': Node.find(registered_project_public_query).count(),
+            'embargoed': Node.find(registered_project_private_query).count(),
+            'withdrawn': Node.find(registered_project_retracted_query).count(),
+        })
+
+    if end_date:
+        totals['keen'] = {'timestamp': end_date.isoformat()}
```

Comment:
Since we have the option of backdating many of these scripts (or parts of all the scripts), does it make sense to add the timestamp to each of them, or just the ones you've done already?